pkg/controller/drain.go

```diff
@@ -179,7 +179,23 @@ func (o *DrainOptions) RunDrain() error {
 		return err
 	}
 
-	err := o.deleteOrEvictPodsSimple()
+	node, err := o.client.CoreV1().Nodes().Get(o.nodeName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		klog.V(4).Info("Node not found: ", o.nodeName)
+		return err
+	} else if err != nil {
+		klog.Errorf("Error getting details for node: %q. Err: %v", o.nodeName, err)
+		return err
+	}
+
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == corev1.NodeReady && condition.Status != corev1.ConditionTrue {
```

Comment:
Does it make sense to enhance this check in order to verify that the node is not ready because the kubelet is not running?

Reply:
I think the `Ready` node-condition represents if kubelet is posting the status or not.
```
  - lastHeartbeatTime: "2020-04-22T04:29:19Z"
    lastTransitionTime: "2020-04-17T14:13:22Z"
    message: kubelet is posting ready status
    reason: KubeletReady
    status: "True"
    type: Ready
```

Or you hinting towards more advanced node-condition/check. We can add new node-condition via NPD say `KubeletHung` and rely on that for force-deletion.

Reply:
No, that was already what I was looking for, makes sense. Then ignore my comment/question above. Thanks!